modern/src/App.js

```diff
@@ -305,11 +305,19 @@ function Layer({ makiObject }) {
   if (img.y !== undefined) {
     params.backgroundPositionY = -Number(img.y);
   }
-  if (img.w !== undefined) {
-    params.width = Number(img.w);
+  if (w !== undefined || img.w !== undefined) {
+    if (w) {
+      params.width = Number(w);
+    } else {
+      params.width = Number(img.w);
+    }
   }
```

Comment:
What about:

```JavaScript
  if (w !== undefined) {
    params.width = Number(w);
  } else if (img.w !== undefined) {
    params.width = Number(img.w);
  }
```

Your code ignores `w` if it's zero. Is that intentional?

Reply:
oh no, that was very unintentional and wrong, and your way would be better anyway hah, thanks!

Reply:
So while that handling of 0 was definitely wrong, it hid a different bug. I've documented the bug in `GuiObject.getwidth()`. I think what we might want to be doing is encoding the logic that the renderer currently has about getting the `width`/`height` to use into `GuiObject.getwidth/height` since it seems to need the same logic which may rely on images...maybe?

Reply:
(so to be clear, I think this is correct, but has introduced another bug thats visible at least in CornerAMP)